pkg/kubelet/kubelet.go

```diff
@@ -2138,9 +2150,15 @@ func (kl *Kubelet) syncLoopIteration(updates <-chan kubetypes.PodUpdate, handler
 			glog.Errorf("Kubelet does not support snapshot update")
 		}
 	case <-kl.resyncTicker.C:
```

Comment:
What if WorkQueue provided the channel instead? It could either work the same as what you have now:

- just move ticker into the WorkQueue
- on each tick look for ready work and send either a slice of ids over the channel, or each id one at a time

Or the work queue could maintain a timer until the next work item is ready, and update it either when a new item is added (with shorter duration), or an item becomes ready.